lib/Parse/ParseDecl.cpp

```diff
@@ -8013,26 +8013,51 @@ ParserStatus Parser::parseGetSet(ParseDeclOptions Flags, ParameterList *Indices,
   bool parsingLimitedSyntax = Flags.contains(PD_InProtocol) ||
                               SF.Kind == SourceFileKind::SIL;
 
+  auto parseImplicitGetter = [&]() {
+    assert(Tok.is(tok::l_brace));
+    accessors.LBLoc = Tok.getLoc();
+    auto *getter = AccessorDecl::createParsed(
+        Context, AccessorKind::Get, storage, /*declLoc*/ Tok.getLoc(),
+        /*accessorKeywordLoc*/ SourceLoc(), /*paramList*/ nullptr,
+        /*asyncLoc*/ SourceLoc(), /*throwsLoc*/ SourceLoc(),
+        /*thrownTy*/ nullptr, CurDeclContext);
+    accessors.add(getter);
+    parseAbstractFunctionBody(getter);
+    accessors.RBLoc = getter->getEndLoc();
+  };
+
   // If the body is completely empty, preserve it. This is at best a getter with
   // an implicit fallthrough off the end.
   if (peekToken().is(tok::r_brace)) {
-    accessors.LBLoc = consumeToken(tok::l_brace);
-    accessors.RBLoc = consumeToken(tok::r_brace);
+    auto consumeEmptyBraces = [&]() {
+      accessors.LBLoc = consumeToken(tok::l_brace);
+      accessors.RBLoc = consumeToken(tok::r_brace);
+    };
 
     // In the limited syntax, fall out and let the caller handle it.
-    if (parsingLimitedSyntax)
+    if (parsingLimitedSyntax) {
+      consumeEmptyBraces();
       return makeParserSuccess();
+    }
 
     if (ResultType != nullptr) {
       // An error type at this point means we couldn't parse
       // the result type for subscript correctly which will be
       // already diagnosed as missing result type in declaration.
-      if (ResultType->getKind() == TypeReprKind::Error)
+      if (ResultType->getKind() == TypeReprKind::Error) {
+        consumeEmptyBraces();
         return makeParserError();
+      }
 
-      diagnose(accessors.RBLoc, diag::missing_accessor_return_decl,
-               /*subscript*/ Indices != nullptr, ResultType);
+      // Otherwise treat this as a 'success' from a parsing perspective. We
+      // need more information to accurately diagnose whether a return is
+      // actually necessary here (e.g. if the empty getter is unreachable,
+      // we admit it).
+      parseImplicitGetter();
+      return makeParserSuccess();
     } else {
+      consumeEmptyBraces();
+
```

Comment:
okay, so to clarify what i think is happening & what you're proposing – this branch seems like it was added via https://github.com/swiftlang/swift/pull/40705 to address this sort of thing:

```swift
var x = true {} // error: unexpected '{' in declaration
```
if i remove the logic, the diagnostics change, but the scenario still appears to be handled in a reasonable manner (to me, anyway). e.g. if the `unexpected_curly_braces_in_decl` diagnostic path is removed (aside: it looks like this is its only use, so it's another diagnostic that can perhaps be removed entirely), then parsing produces:

```swift
var x = true {} // error: computed property must have an explicit type
```

if we add a type annotation, given the changes proposed in this PR, we successfully parse the code, but typechecking fails with:

```swift
var x: Bool = true {} // error: variable with getter/setter cannot have an initial value
```
so given the other suggested branch removal, the final logic for this conditional would be something like:

1. if we're parsing in the limited syntax mode, consume the empty braces
2. otherwise, parse an implicit getter
3. in both cases, return a parser success

locally this appeared to pass tests other than the affected error message change mentioned above. if this sounds accurate to you, i'll try to clean up this code and update the tests.